cdhweb/people/models.py

```diff
@@ -371,70 +381,115 @@ def current_title(self):
         return self.user.current_title
 
 
+class ProfilePage(Page):
+    """Profile page for a Person, managed via wagtail."""
+    person = models.OneToOneField(
+        Person, help_text="Corresponding person for this profile",
+        null=True, on_delete=models.SET_NULL)
+    image = models.ForeignKey('wagtailimages.image', null=True,
+                              blank=True, on_delete=models.SET_NULL,
+                              related_name='+')  # no reverse relationship
+    education = RichTextField(features=PARAGRAPH_FEATURES, blank=True)
+    bio = StreamField(BodyContentBlock, blank=True)
+
+    # admin edit configuration
+    content_panels = Page.content_panels + [
+        FieldRowPanel(
+            (FieldPanel("person"), ImageChooserPanel("image")), "Person"),
+        FieldPanel("education"),
+        StreamFieldPanel("bio")
+    ]
+
+    parent_page_types = ["people.PeopleLandingPage"]
+    subpage_types = []
+
+    def get_context(self, request):
+        """Add recent BlogPosts by this Person to their ProfilePage."""
+        context = super().get_context(request)
+
+        # get 3 most recent published posts with this person as author
+        recent_posts = BlogPost.objects.filter(
+            users__id=self.person.user.id).published()[:3]
+
+        # add to context and set open graph metadata
+        context.update({
+            "opengraph_type": "profile",
+            "recent_posts": recent_posts
+        })
+
+        return context
+
+    def clean(self):
+        """Validate that a Person was specified for this profile."""
+        # NOTE we can't cascade deletes to wagtail pages without corrupting the
+        # page tree. Instead, we use SET_NULL, and then add a delete handler
+        # to Person to delete the corresponding profile manually. We still need
+        # to make sure that it's impossible to create a ProfilePage without a
+        # corresponding Person. More info:
+        # https://github.com/wagtail/wagtail/issues/1602
+        if not self.person:
+            raise ValidationError("Profile page must be for existing person.")
+
+
+class PeopleLandingPage(LandingPage):
+    """LandingPage subtype for People that holds ProfilePages."""
+    # NOTE this page can't be created in the page editor; it is only ever made
+    # via a script or the console, since there's only one.
+    parent_page_types = []
+    # NOTE the only allowed child page type is a ProfilePage; this is so that
```

Comment:
Ah. I wondered why we needed a subclass of the landing page. Seems reasonable, I guess, since we don't want profile pages being added elsewhere (I think).